server/src/contact/contact.service.ts

```diff
@@ -0,0 +1,29 @@
+import { 
+  Injectable,
+} from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Contact } from './contact.entity';
+
+@Injectable()
+export class ContactService {
+  constructor(
+    @InjectRepository(Contact)
+    private readonly contactRepository: Repository<Contact>,
+  ) {}
+
+  activeContacts() {
+    return this.contactRepository.createQueryBuilder('active_contacts')
+      .where('active_contacts.statuscode = :statuscode', { statuscode: 'Active' });
+  }
+
+  async findOne(opts: any): Promise<Contact> {
+    return this.contactRepository.findOneOrFail(opts);
+  }
+
+  async findByEmail(email): Promise<Contact> {
+    return this.activeContacts()
+      .andWhere('emailaddress1 ILIKE :email', { email })
+      .getOne();
+  }
+}
```

Comment:
None of this would be necssary or woudl work because we aren't using a SQL data ase for this. THis is all TypeORM stuff which is specific to SQL datbases, not WEb API